imsim/stamp.py

```diff
@@ -84,6 +84,27 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
         Returns:
             xsize, ysize, image_pos, world_pos
         """
+
+        # First do a parsing check to make sure that all the options passed to
+        # the config are valid, and no required options are missing.
+
+        req = {'fft_sb_thresh': float,
+               'band': str,
+               'world_pos': galsim.CelestialCoord,
+               'diffraction_psf': dict
+               }
+
+        opt = {'airmass': dict,
+               'rawSeeing': float,
+               'max_flux_simple': float,
+               'draw_method': str,
+               'maxN': int,
+               'size': int,
+               'photon_ops': dict
```

Comment:
Some of these are in ignore, so you don't need to include them (They are standard GalSim optional items, which GalSim has already put into ignore here.): draw_method, maxN, size, photon_ops, and world_pos (from above).